feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsViewModel.kt

```diff
@@ -0,0 +1,130 @@
+import com.chocolatecake.viewmodel.home.TopRatedUiState
+import com.chocolatecake.viewmodel.people.mapper.MoviesByPeopleUiMapper
+import com.chocolatecake.viewmodel.people.mapper.PeopleDataUiMapper
+import com.chocolatecake.viewmodel.people.mapper.TvShowsByPeopleUiMapper
+import com.chocolatecake.viewmodel.search.MovieUiMapper
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class PeopleDetailsViewModel @Inject constructor(
+    private val getPeopleDetailsUseCase: GetPeopleDetailsUseCase,
+    private val getMoviesByPeopleUseCase: GetMoviesByPeopleUseCase,
+    private val getTvShowsByPeopleUseCase: GetTvShowsByPeopleUseCase,
+    private val peopleDataUiMapper: PeopleDataUiMapper,
+    private val moviesByPeopleUiMapper: MoviesByPeopleUiMapper,
+    private val tvShowsByPeopleUiMapper: TvShowsByPeopleUiMapper
+
+) : BaseViewModel<PeopleDetailsUiState, PeopleDetailsUiEvent>(PeopleDetailsUiState()),
+    PeopleDetailsListener {
+
+    var num_people_movies = ""
+
+    fun getPersonData(personId: Int) {
+        viewModelScope.launch {
+            try {
+                val data = getPeopleDetailsUseCase(personId)
+                val peopleData = peopleDataUiMapper.map(data)
+                onSuccessGetPersonData(peopleData)
+            } catch (e: Throwable) {
+                onErrorGetPersonData(e)
+            }
+
+        }
+
+    }
+
+    private fun onSuccessGetPersonData(peopleDataUiState: PeopleDetailsUiState.PeopleDataUiState) {
+        _state.update {
+            it.copy(
+                peopleData = peopleDataUiState,
+                isLoading = false
+            )
+    }
+        Log.e("num_people_movies",num_people_movies)  }
+
+    private fun onErrorGetPersonData(e: Throwable) {
+        val errors = _state.value.onErrors.toMutableList()
+        errors.add(e.message.toString())
+        _state.update { it.copy(onErrors = errors, isLoading = false) }
+    }
+
+    fun getMoviesByPeople(personId: Int) {
+        tryToExecute(
+            call =  {getMoviesByPeopleUseCase.invoke(personId)} ,
+            onSuccess = ::onSuccessGetMoviesByPeople,
+            onError = ::onErrorGetMoviesByPeople,
+            mapper = moviesByPeopleUiMapper
+        )
+    }
+
+
+    private fun onSuccessGetMoviesByPeople(moviesUiState: List<MediaVerticalUIState>) {
+        _state.update {
+            it.copy(
+                Movies = moviesUiState,
+                isLoading = false,
+            )
+        }
+        num_people_movies=moviesUiState.size.toString()
+        Log.e("num_people_movies",num_people_movies)
+    }
+
+    private fun onErrorGetMoviesByPeople(e: Throwable) {
+        val errors = _state.value.onErrors.toMutableList()
+        errors.add(e.message.toString())
+        _state.update { it.copy(onErrors = errors, isLoading = false) }
+    }
```

Comment:
```suggestion
    private fun onError(th: Throwable) {
        val errors = _state.value.errors.toMutableList()
        errors.add(th.message.toString())
        _state.update { it.copy(errors = errors, isLoading = false) }
    }
```
most viewModels in the project uses this implementation

Reply:
what you mean

Reply:
what you mean